Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -106,7 +173,7 @@
     /**
      * Random number generator used by the forest.
      */
-    protected final Random rng;
+    protected Random random;
```

Comment:
Is removing `final` so you can do full state restoration in serde?

Reply:
yes (mostly and some no :)). The full state restoration goes via the randomSeed (which is how the forest instantiates the trees). But for the tests -- they mock up a Random() and test the different codepath (which is very useful). So the removal of final is because we keep both the seed and the Random(). That being said - this was happening because there were a thousand constructors, which is now changed thx to Josh's last PR and they all go through the builders. Maybe this is a cleanup we can have post merge.

Reply:
Thanks, got it. Not to mention that there may be a use case to reinitializing the rng in an existing model.

Reply:
Yes - it may be useful to rerandomize (or the opposite, sync) across different forest :) The models encapture stochastic process states, so these operations may see use via the dynamic setConfig options in a 2.X